### ISSUE TITLE
Define content encoding headers for 3rd party mirrors

### ISSUE THREAD

**Issue by Tiippex** [isaaceindhoven/velocita-proxy#8]
Hey,

gitlab.com has adjusted the API interface with one of the latest releases. If no Accept-Encoding header is defined, GitLab uses the Content-Encoding header with the value "br", which is used by Composer or libcurl.

curl error 61 while downloading **URL**: Unrecognized content encoding type. libcurl understands deflate, gzip content encodings.

If gitlab.com receives the Accept-Encoding header with the value gzip, then the correct Content-Encoding header with gzip is also returned.

GitLab:
![image](https://user-images.githubusercontent.com/16509128/195367643-f6cc7529-d9d9-4e59-8574-546694ae1caa.png)
Response Header:
![image](https://user-images.githubusercontent.com/16509128/195367677-0edb3297-cde5-464b-95e2-daf857c43505.png)

Velocita -> Gitlab:
![image](https://user-images.githubusercontent.com/16509128/195367925-440ad684-408e-4b4a-99ef-bcfe7174c361.png)
Response Header:
![image](https://user-images.githubusercontent.com/16509128/195367950-5395ac51-62fe-4f7e-8f55-509747475256.png)

Composer error:
![image](https://user-images.githubusercontent.com/16509128/195370049-21842730-d2fb-42e7-9122-ee5607d6e6d1.png)

Could you help me with this issue, please?

Regards
Max

**Comment by GMTA** [comment]
Hi @Tiippex,

Can you please provide the configuration for GitLab that you are using in Velocita? You can anonymize the paths if you'd like. Can you please also provide me with the full output of a `composer install -vvv` run?

Velocita does not interject downloading packages, it uses Composer's own mechanism for downloading files. So I'm a bit surprised that this issue is popping up; I'd expect Composer to fail as well.

**Comment by GMTA** [comment]
Hm, this might be caused by the value of `Accept-Encoding` missing in `proxy_cache_key`. Need to look into this.

**Comment by Tiippex** [comment]
> Hi @Tiippex,
> 
> Can you please provide the configuration for GitLab that you are using in Velocita? You can anonymize the paths if you'd like. Can you please also provide me with the full output of a `composer install -vvv` run?
> 
> Velocita does not interfere with downloading files, it uses Composer's own mechanism for downloading. So I'm a bit surprised that this issue is popping up; I'd expect Composer to fail as well.

Hey sure,

For every dump I cleared the composer cache and removed the vendor folder.

Composer.json
https://paste.tiippex.de/1gz1WfKFJHSAsVMdiLuOcR

With velocita: 
https://paste.tiippex.de/6KD8jzbWv4YM0t7VGbFhPE

Velocita .env:
https://paste.tiippex.de/5ZQF56luLg9g6GIs8imhFF

Thanks!
Max

**Comment by GMTA** [comment]
As a test, I set up a GitLab Composer package registry and performed a `curl` call to it:

```
curl --header 'Accept-Encoding: br, gzip, deflate' --verbose --user 'xxx:yyy' -X GET https://gitlab.com/api/v4/group/1337/-/packages/composer/packages
...
< HTTP/2 200
...
< vary: Origin
< content-encoding: br
```

The nice thing for GitLab to do here is respond with `Vary: Origin, Accept-Encoding` at the very least. But they don't, so we need to figure out the best way to encode the `Accept-Encoding` into the cache key.

**Comment by GMTA** [comment]
Hi @Tiippex, I've included a fix in release 3.5.3. Can you please try running that Docker image version and let me know if that resolves your issue?

**Comment by Tiippex** [comment]
Hey @GMTA, looks good now!
I wasn't able to reproduce the problem with 3.5.3. 

Thank you very much for the fast fix!

Regards
Max
